src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -1090,4 +1097,73 @@ public function setEchoFormat($format)
     {
         $this->echoFormat = $format;
     }
+
+    /**
+     * Compile the switch statements into valid PHP.
+     * @param  string  $variable
+     *
+     * @return string
+     */
+    public function compileSwitch($variable)
+    {
+        return "<?php switch({$variable}){ ";
+    }
+
+    /**
+     * Compile the first case statement into valid PHP.
+     * @param  string  $value
+     *
+     * @return string
+     */
+    private function compileFirstCase($value)
+    {
+        return "case {$value}:  ?>";
```

Comment:
So the expected syntax would be:

```
@switch($num)
@firstcase(10)
```

However i guess that this will break if there's html between ``` @switch ``` and ``` @firstcase ```.
Also this requires the use of ``` @firstcase ```
IMO we could remove this and add a closing tag to the ``` compileSwitch() ``` this way we could use ``` @case ``` wherever we want, not sure if there's a reason for this syntax.